agent/utils/ticker.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"context"
+	"math/rand"
+	"time"
+)
+
+// JitteredTicker works like a time.Ticker except with randomly distributed ticks
+// between start and end duration.
+func JitteredTicker(ctx context.Context, start, end time.Duration) <-chan time.Time {
```

Comment:
nit, naming: can we name this `NewJitteredTicker`? since stdlib uses `NewTicker`